keps/sig-network/4559-kubelet-probes/README.md

```diff
@@ -0,0 +1,1037 @@
+[future successor to NetworkPolicy]: https://github.com/kubernetes-sigs/network-policy-api/issues/136
+[multi-tenant scenarios]: https://github.com/kubernetes/kubernetes/issues/31824
+[Multi-network feature]: https://github.com/kubernetes/enhancements/issues/3698
+[kubernetes #101324]: https://github.com/kubernetes/kubernetes/issues/101324
+[the NetworkPolicy has not yet been fully programmed]: https://kubernetes.io/docs/concepts/services-networking/network-policies/#pod-lifecycle
+[readiness gates]: https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-readiness-gate
+[Exec probes are much, much slower than network probes]: https://medium.com/netcracker/exec-probes-a-story-about-experiment-and-relevant-problems-12de616c0a76
+[kubernetes #89898]: https://github.com/kubernetes/kubernetes/issues/89898
+[kubernetes #115143]: https://github.com/kubernetes/kubernetes/pull/115143
+[kubernetes #102613]: https://github.com/kubernetes/kubernetes/issues/102613
+[kubevirt]: https://github.com/kubevirt/kubevirt/blob/main/docs/devel/networking.md#vmi-networking
+
+## Motivation
+
+### Goals
+
+```
+<<[UNRESOLVED goals ]>>
+
+The exact details of the goals depend on whether we decide to
+deprecate the existing probe types, or just change their semantics.
+
+Also, there may be additional goals depending on the resolution of the
+UNRESOLVED section in the Summary.
+
+<<[/UNRESOLVED]>>
+```
+
+- Continue to backward-compatibly support all existing probes by
+  default (at least for a while).
+
+- Deprecate the [`TCPSocketAction.Host`] and [`HTTPGetAction.Host`]
+  fields in pod probes (since they allow the SSRF attack and have no
+  other compelling use case).
+
+    - Allow administrators to use [Pod Security admission] to block
+      Pods that have probes using those fields.
+
+- Move to a system where TCP, HTTP, and GRPC pod probes are run inside
+  the pod network namespace rather than being sent over the pod
+  network, so that (among other things) kubelet does not need to have
+  any particular pod-network security privileges.
+
+    - (For pods that really have a strong use case for checking
+      connectivity in some non-standard way, such as via a load
+      balancer IP, it would still be possible for a user to write an
+      exec probe that uses `curl` or the like to connect to that IP
+      from the pod itself.)
+
+    - If we implement this by creating a new set of probe types,
+      rather than by modifying the semantics of the existing probes,
+      then it should be possible to block the old probe types via Pod
+      Security admission, so that administrators can create
+      AdminNetworkPolicies that don't include the kubelet probe hole.
+
+- Allow dual-stack / wrong-single-stack probing, either based on
+  information in the probe definition (e.g. `ipFamily: IPv6` or
+  `ipFamilyPolicy: RequireDualStack`) or by always doing "[Happy
```

Comment:
I would not say that happy eyeballs is a successful protocol , it also had to go through a new version https://datatracker.ietf.org/doc/html/rfc8305 and is very unreliable and causes a lot of issues in production environments